hotelservice/Services/Hotel/HotelService.cs

```diff
@@ -87,6 +87,36 @@ public async Task<GetHotelResponse> GetHotel(GetHotelRequest request)
 
     public async Task<HotelBookRoomsResponse> BookRooms(HotelBookRoomsRequest request)
     {
+        
+        var hotelId = request.BookingDetails.Id;
+
+        var popularHotel = await _dbContext.PopularHotels
+            .FirstOrDefaultAsync(h => h.Id == hotelId);
+        
+        var popularHotelInfo = await _dbContext.Hotels
+            .FirstOrDefaultAsync(h => h.Id == hotelId);
+
+        
+        if (popularHotel == null && popularHotelInfo != null)
+        {
+            // Hotel does not exist, create a new record
+            popularHotel = new PopularHotel
+            {
+                Id = hotelId,
+                Name = popularHotelInfo.Name,
+                City = popularHotelInfo.City,
+                Country = popularHotelInfo.Country,
+                Counter = 1 
+            };
+            _dbContext.PopularHotels.Add(popularHotel);
+        }
+
+        else
+        {
+            popularHotel.Counter++;
+            _dbContext.PopularHotels.Update(popularHotel);
+        }
+
```

Comment:
All this code should be added at the end of this function (preferably just before changes are added to the database), so that popularity stats are updated only if booking was successful

Reply:
Fixed, added it at the end of try block.